src/modules/date/date-string.transformations.ts

```diff
@@ -29,4 +29,24 @@ export class DateStringTransformations {
     }
     return dateOnlyString + 'T00:00:00Z';
   }
+
+  getEffectiveDate(dateOnlyString: DateString): DateString {
+    const today = new Date();
+    if (new Date(dateOnlyString) < today) {
+      return this.addTimeToDateOnlyString(dateOnlyString);
+    } else {
+      const todayAsDateOnlyString = this.removeTimeIfExists(today.toISOString());
+      return this.addTimeToDateOnlyString(todayAsDateOnlyString);
+    }
+  }
+
+  getExpirationDate(dateOnlyString: DateString | null): DateString | null {
+    if (dateOnlyString) {
+      return this.addTimeToDateOnlyString(dateOnlyString);
+    } else if (PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate) {
+      return PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate;
+    } else {
+      return null;
+    }
```

Comment:
I don't think we should check whether or not `PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate` is truthy here, we should just use it.
So
```
if (dateOnlyString) {
  return this.addTimeToDateOnlyString(dateOnlyString);
}

return PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate;
```

Reply:
Agree with Luke that PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate will always exist.

Date module is generic helper, most likely it shouldn't know about Deal investors, Facilities and other ACBS record types.  Maybe we can pass default value as input parameter.

Reply:
Good point about the date module being generic @avaitonis. I decided to move the logic to the deal investor service itself, as well as removing the truthy check. The logic now looks like this:

```
const expiryDateOnlyString = newInvestor.expiryDate;
    const expirationDateString = expiryDateOnlyString
      ? this.dateStringTransformations.addTimeToDateOnlyString(expiryDateOnlyString)
      : PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate;
```